hal/inc/ble_hal.h

```diff
@@ -273,13 +285,37 @@ typedef struct hal_ble_att_mtu_updated_evt_t {
     size_t att_mtu_size;
 } hal_ble_att_mtu_updated_evt_t;
 
+typedef struct hal_ble_pairing_request_evt_t {
+    // We may expose the LESC, OOB, bond flags set by peer when necessary.
+    uint8_t reserved[4];
+} hal_ble_pairing_request_evt_t;
+
+typedef struct hal_ble_pairing_passkey_display_evt_t {
+    const uint8_t* passkey;
+} hal_ble_pairing_passkey_display_evt_t;
+
+typedef struct hal_ble_pairing_passkey_input_evt_t {
+    uint8_t reserved[4];
+} hal_ble_pairing_passkey_input_evt_t;
+
+typedef struct hal_ble_pairing_status_updated_evt_t {
+    int status;
```

Comment:
`int32_t`?

Reply:
The SoftDevice features `uint8_t` for the authentication status and the `status` here will also indicate system error, so I think `int` is sufficient and is consistent with the function that returns system error.